src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -145,8 +154,8 @@
     private Optional<DatabaseChangeMonitor> changeMonitor = Optional.empty();
     private JabRefExecutorService executorService;
 
-    public BasePanel(JabRefFrame frame, BasePanelPreferences preferences, BibDatabaseContext bibDatabaseContext, ExternalFileTypes externalFileTypes) {
-        this.preferences = Objects.requireNonNull(preferences);
+    public BasePanel(JabRefFrame frame, BasePanelPreferences basePanelPreferences, BibDatabaseContext bibDatabaseContext, ExternalFileTypes externalFileTypes, GroupViewMode groupViewMode, FilePreferences filePreferences, ImportFormatPreferences importFormatPreferences, UpdateFieldPreferences updateFieldPreferences, FileUpdateMonitor fileUpdateMonitor, StateManager stateManager, Consumer<ColumnPreferences> updateColumnPreferences, Supplier<TimestampPreferences> timestampPreferencesSupplier) {
```

Comment:
I think it's better to pass a PreferenceService if you need to store preference options (or get the current preferences at a later moment).

Reply:
Oh, yeah. - I wonder why we don't more often use this service in the GUI instead of relying on JabRefPreferences directly.

Reply:
Because it was introduced only recently.